# deSRA
**d**ifferential **e**xpression of **SRA**: An automated protocol to extract differences in gene expression from public NGS datasets and perform comparative analyses through an R Shiny GUI.

For more information on deSRA, see: https://github.com/NCBI-Hackathons/deSRA. The purpose of this repository is to provide new code and documentation for a GUI on top of deSRA.

## Introduction
The future of biomedical research depends on the ability to rapidly access and analyze Next-Generation Sequencing (NGS) data stored at the NCBIâ€™s Sequence Read Archive (SRA). NGS provides an unprecedented level of resolution, allowing researchers to ask previously unanswerable questions such as how cancer pathogenesis might be mediated by very small changes in gene expression.


With ~3 million records currently stored in the SRA database, and submissions growing exponentially, the SRA collection represents a treasure-trove of data to be mined by academia and industry. The information contained in a single SRA dataset is equivalent to hundreds of in vitro experiments worth of work, potentially saving thousands of dollars and research hours.


Here we present deSRA, a tool for processing and visualizing RNA-seq data from SRA datasets on the fly. deSRA facilitates the interrogation of SRA datasets for differential gene expression via dockerized pipeline. deSRA makes it easy for biologists to perform pathway enrichment analyses and to interrogate which genes are differentially expressed between experiments.


![alt text](https://user-images.githubusercontent.com/12971527/32574712-ff0dbfb8-c49f-11e7-8404-1e209a51c5c0.png "Overview Diagram")



# Future Development Plans


# Project team  

deSRA team:   
John Garner  
Mingzhang Yang  
Eneida Hatcher  
Anna-Leigh Brown  
Peter Meric  
Roberto Vera Alvarez  
Ryan Connor  
Nora Husain  
Aynex Mercado  

GUI team:  
Kristin Valdez  
Robert Guthridge  
Ryan Connor  
Luli Zou
